### DISCUSSION TITLE
Run Bash Script

### DISCUSSION THREAD

**Discussion by sulaiman35** [concourse/concourse#8148] [Help & Support]
Hi,

I wanted to execute a bash script in Task and it will gather some results, but I want to fail if the result is not as expected, so need guidance to where should I put that bash script and how I should fail or raise notification if I don't get the result of my need.

Please confirm where I should put the in Tasks, or in any other location ?

Regards,
Muhammad Sulaiman

**Comment by DacoDev** [comment]
I've used the below to run sh, if the image you pull has bash you'd be able to replace sh with bash. For the errors you'll probably just need to add that logic into your script, if it doesn't exit 0 it would fail that task.

```
jobs:
- name: some-job
  plan:
  - task: extract
    config:
      platform: linux
      image_resource:
        type: docker-image
        source: {repository: busybox}
      run:
        path: sh
        args:
        - -c
        - |
          env
          foo=y
          if [ "${foo}" == 'x' ];
            then
              echo "Success"
            else
              echo "Fail"
              exit 1
          fi
```
